azext_iot/operations/events3/_events.py

```diff
@@ -137,15 +147,21 @@ def _output_msg_kpi(msg):
         if devices and origin not in devices:
             return
 
+        # TODO: Super shortcut for PnP
+        if pnp_context:
+            msg_interface_id = str(msg.annotations.get(b'iothub-interface-id'), 'utf8')
+            if not msg_interface_id:
+                return
+
+            if interface_id:
+                if msg_interface_id != interface_id:
+                    return
+
         event_source = {'event': {}}
 
         event_source['event']['origin'] = origin
 
-        payload = ''
-
-        data = msg.get_data()
-        if data:
```

Comment:
What's deleted here is the newer pattern. `payload = str(next(msg.get_data()), 'utf8')` was the older one. There is a chance iterating msg.get_data() could return None.

Reply:
reverted